pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/TopicTransactionBuffer.java

```diff
@@ -509,6 +509,16 @@ public PositionImpl getMaxReadPosition() {
         }
     }
 
+    @Override
+    public CompletableFuture<Position> getLastCanDispatchPosition() {
+        PositionImpl tnxMaxReadPosition = getMaxReadPosition();
+        if (topic.getLastPosition() == tnxMaxReadPosition) {
```

Comment:
I know `tnxMaxReadPosition` and `topic.getLastPosition()` are the same object if they are equals to, but for the compare about two `PositionImpl` objects, `topic.getLastPosition.compareTo(tnxMaxReadPosition) == 0` is better